newton/_src/sim/builder.py

```diff
@@ -483,6 +516,69 @@ def __init__(self, up_axis: AxisType = Axis.Z, gravity: float = -9.81):
         self.equality_constraint_polycoef = []
         self.equality_constraint_key = []
         self.equality_constraint_enabled = []
+        # Custom attributes (user-defined per-frequency arrays)
+        self.custom_attributes: dict[str, CustomAttribute] = {}
+
+    @staticmethod
+    def _default_for_dtype(d: object) -> Any:
+        """Get default value for dtype when not specified."""
+        # quaternions get identity quaternion
+        if d is wp.quat:
+            return wp.quat_identity()
+        # vectors default to zeros of their length
+        if wp.types.type_is_vector(d):
```

Comment:
Warp vectors are already zero when you create them. I think the only case that requires any custom default is the `wp.quat` dtype.